### PR TITLE
Comment operand.rs

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

This PR is 95% comments, but the diff might be hard to read there because some code really needed moving around. All function addition/removals are actually just moves. I left their bodies alone except once where I added a comment.

There were two meaningful changes:

1. A couple of function had `pub` removed because they didn't need it.
2. I changed the "unsupported feature" message for const slices: 

```diff
+                        let operation_name = format!("Constant slice for type {}", slice_ty);
                         return self.codegen_unimplemented_expr(
-                            "Constant slice value with 2+ bytes",
+                            &operation_name,
```

Everything else should largely be adding doc comments to functions, so kinda skipping through the diff to see what I'm adding there is the easiest way to proceed.

### Call-outs:

Some of this code looks like it could use revising...

### Testing:

* How is this change tested? ci

* Is this a refactor change? yes, no behavior change

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs [hunk 1]
```diff
@@ -23,6 +23,9 @@ enum AllocData<'a> {
 }
 
 impl<'tcx> GotocCtx<'tcx> {
+    /// A MIR operand is either a constant (literal or `const` declaration) or a place
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:26]: Here is an example that I think you should first document what `codegen_operand` does. After that you can add more details on what the operand argument actually is.

**Review by celinval — COMMENTED** [review]
Thank you!!

Can you just please make sure that you always start the documentation with a statement that says what the function is supposed to do? Here is a good ref: https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs (hunk 1 above)
Comment (tedinski) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:26]: I tend to leave that out when the name (and maybe type) is really good. What would you like to say about this function besides, uh, "codegens an operand" :)

Comment (celinval) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:26]: Something like "generate a goto expression that represents a MIR operand."

**Review by celinval — APPROVED** [review]

**Pull request merged**